src/Sulu/Bundle/SecurityBundle/Controller/RoleController.php

```diff
@@ -301,6 +301,8 @@ public function putAction(Request $request, $id)
             }
         } catch (EntityNotFoundException $enfe) {
             $view = $this->view($enfe->toArray(), 404);
+        } catch (DoctrineUniqueConstraintViolationException $e) {
+            $view = $this->view((new RestException('Entry already Exists!', 409))->toArray(), 409);
```

Comment:
I know I've explained you something like this, but when looking at it I recognized a better solution: Can you create a separate `RoleNameAlreadyExistsException`, and simply throw it in here instead of creating this `$view` variable?

Then we could handle the error using http://symfony.com/doc/master/bundles/FOSRestBundle/4-exception-controller-support.html and https://symfony.com/doc/current/bundles/prepend_extension.html.

Just come back to me if you have any questions.